pkg/odo/cli/component/create.go

```diff
@@ -263,8 +282,50 @@ func (co *CreateOptions) setResourceLimits() error {
 	return nil
 }
 
+// Validate if devfile component already exists
+func validateDevfileComponentCreateRequest() error {
+	if util.CheckPathExists(DevfilePath) {
+		return errors.New("Devfile.yaml already exists")
+	}
+
+	if util.CheckPathExists(EnvFilePath) {
+		return errors.New("env.yaml already exists")
```

Comment:
What happens if the devfile.yaml is missing but there is already an env.yaml?

Reply:
I think we should not overwrite devfile if its present in the local disk. If user devfile is present, we should continue with that!

Reply:
My original thinking was we should error out if any file is already exists. I guess your point is if devfile.yaml is missing but there is already an env.yaml we should pull down the devfile.yaml according to the existing env.yaml? If that's your point we also need to consider if there is a conflict between the new arguments that user pass in via `odo create` and the existing env.yaml. So follow the existing design as odo1 and keep it simple, we only generate two files if they are not exist, otherwise we error out.

But I think I still need to modify the code to cover all cases and throw proper errors, I think I should do:

Case | Current version | Modified version
------------|------------ | -------------
Both are missing | Create both files | Create both files
Only devfile.yaml is already exists | Throw error that devfile.yaml already exists | Throw error that devfile.yaml already exists
Only env.yaml is already exists | Throw error that env.yaml already exists | Throw error that env.yaml already exists
Both are already exist | Throw error that devfile.yaml already exists | **Throw errors that both files are already exist**

Reply:
@maysunfaisal Since we don't if user wants to overwrite or still use the existing devfile, so I would say we can leave the option to user and let user decide which way is better, so that's why I just directly return the error that devfile is already exists (same for existing odo, odo won't overwrite user's previous configuration).